apps/condo/pages/auth/register.tsx

```diff
@@ -1,120 +1,52 @@
-
-import { useIntl } from '@core/next/intl'
-import { Col, Row, Typography } from 'antd'
-import Router from 'next/router'
-import React, { useContext, useEffect, useState } from 'react'
-import { ButtonHeaderAction } from '@condo/domains/common/components/HeaderActions'
+import { useRouter } from 'next/router'
+import { useCallback, useEffect } from 'react'
 import { useMutation } from '@core/next/apollo'
 import { runMutation } from '@condo/domains/common/utils/mutations.utils'
-import { getClientSideSenderInfo } from '@condo/domains/common/utils/userid.utils'
+import { useIntl } from '@core/next/intl'
 import { CREATE_ONBOARDING_MUTATION } from '@condo/domains/onboarding/gql'
-import { fontSizes } from '@condo/domains/common/constants/style'
+import { RegisterLayout } from '@condo/domains/user/components/auth/RegisterLayout'
+import { useRegisterContext } from '@condo/domains/user/components/auth/RegisterContextProvider'
 
-const RegisterPage: AuthPage = () => {
+export default function RegisterPage () {
     const intl = useIntl()
-    const RegistrationTitleMsg = intl.formatMessage({ id: 'pages.auth.RegistrationTitle' })
-    const PhoneConfirmTokenErrorLabel = intl.formatMessage({ id: 'pages.auth.register.PhoneConfirmTokenErrorLabel' })
-    const PhoneConfirmTokenErrorMessage = intl.formatMessage({ id: 'pages.auth.register.PhoneConfirmTokenErrorMessage' })
-    const RestartPhoneConfirmLabel = intl.formatMessage({ id: 'pages.auth.register.RestartPhoneConfirmLabel' })
-
-    const { token, isConfirmed, tokenError, setToken, setTokenError } = useContext(RegisterContext)
-    const [state, setState] = useState('inputPhone')
-
+    const { token, tokenLoading, isConfirmed, setToken } = useRegisterContext()
+    const router = useRouter()
     const [createOnBoarding] = useMutation(CREATE_ONBOARDING_MUTATION, {
         onCompleted: () => {
-            Router.push('/onboarding')
+            router.push('/onboarding')
         },
     })
 
-    const initOnBoarding = (userId: string) => {
+    const initOnBoarding = useCallback((userId: string) => {
         const onBoardingExtraData = {
             dv: 1,
             sender: getClientSideSenderInfo(),
         }
 
         const data = { ...onBoardingExtraData, type: 'ADMINISTRATOR', userId }
 
-        return runMutation({
+        runMutation({
             mutation: createOnBoarding,
             variables: { data },
             intl,
         })
-    }
+    }, [createOnBoarding, intl])
+
+    const onRegisterFinish = useCallback((userId: string) => {
+        setToken(null)
+        initOnBoarding(userId)
+    }, [initOnBoarding, setToken])
 
     useEffect(() => {
-        if (token && isConfirmed) {
-            setState('register')
-        } else if (token) {
-            setState('validatePhone')
-        } else {
-            setState('inputPhone')
+        if (!tokenLoading && (!token || !isConfirmed)) {
+            router.push('/auth/input')
```

Comment:
Please, use more meaningful route name.
The name should tell what should be done in corresponding page.

Reply:
I guess, it redirects to page where a user starts identity confirmation, so, the name can be `/auth/start-confirmation`.